tests/test_flow/fakes.py

```diff
@@ -67,132 +67,111 @@ def submit(self, task_config, fn, *args, **kwargs):
         ).submit()
 
 
-class FakeBlob:
-    def __init__(self, bucket, name, blobs):
-        assert not name.startswith("/")
-        self.bucket = bucket
-        self.name = name
-        self.blobs = blobs
-
-    def upload_from_string(self, data):
-        self.blobs[self._url] = data
-
-    def download_as_string(self):
-        return self.blobs[self._url]
-
-    def upload_from_filename(self, filename):
-        self.upload_from_string(open(filename, "r").read())
-
-    def download_to_filename(self, filename):
-        open(filename, "w").write(self.download_as_string())
-
-    def exists(self):
-        return self._url in self.blobs
-
-    def delete(self):
-        del self.blobs[self._url]
-
-    @property
-    def _url(self):
-        return f"gs://{self.bucket}/{self.name}"
-
-
-class FakeBucket:
-    def __init__(self, name, blobs):
-        self.name = name
-        self.blobs = blobs
-
-    def blob(self, blob_name):
-        return FakeBlob(self.name, blob_name, self.blobs)
-
-    def get_blob(self, blob_name):
-        return self.blob(blob_name)
-
-    def list_blobs(self, prefix):
-        assert not prefix.startswith("gs://")
-        return [
-            FakeBlob(self.name, k.replace(f"gs://{self.name}/", "", 1), self.blobs)
-            for (k, v) in self.blobs.items()
-            if k.startswith(f"gs://{self.name}/{prefix}")
-        ]
-
-
-class FakeGCS:
+# TODO It would be nice to use a different fsspec implementation here instead
+# of writing a custom one. We tried fsspec.implementations.memory.MemoryFileSystem
+# implementation, but that has a few problems.
+# 1) The glob method returns the protocol gs:// whereas gcsfs implementation doesn't
+# 2) There is a bug in the `rm` method and the tests fails to wipe the gcs cache.
```

Comment:
I opened an issue in fsspec for this bug as well https://github.com/intake/filesystem_spec/issues/449